source/extensions/dynamic_modules/abi.h

```diff
@@ -2874,6 +2874,199 @@ bool envoy_dynamic_module_callback_network_filter_is_ssl(
+ * @param sans_out is the output pointer to an array of SAN strings. Memory is owned by Envoy and
+ * valid until the end of the callback.
+ * @param sans_count_out is the output pointer to the number of SANs in the array.
+ * @return true if SSL is available and SANs were retrieved, false otherwise.
+ */
+bool envoy_dynamic_module_callback_network_filter_get_ssl_uri_sans(
+    envoy_dynamic_module_type_network_filter_envoy_ptr filter_envoy_ptr,
+    envoy_dynamic_module_type_envoy_buffer** sans_out, size_t* sans_count_out);
+
+/**
+ * envoy_dynamic_module_callback_network_filter_get_ssl_dns_sans is called by the module to get
+ * the DNS Subject Alternative Names from the peer certificate.
+ *
+ * @param filter_envoy_ptr is the pointer to the DynamicModuleNetworkFilter object.
+ * @param sans_out is the output pointer to an array of SAN strings. Memory is owned by Envoy and
+ * valid until the end of the callback.
+ * @param sans_count_out is the output pointer to the number of SANs in the array.
+ * @return true if SSL is available and SANs were retrieved, false otherwise.
+ */
+bool envoy_dynamic_module_callback_network_filter_get_ssl_dns_sans(
+    envoy_dynamic_module_type_network_filter_envoy_ptr filter_envoy_ptr,
+    envoy_dynamic_module_type_envoy_buffer** sans_out, size_t* sans_count_out);
+
+/**
+ * envoy_dynamic_module_callback_network_filter_get_ssl_subject is called by the module to get
+ * the subject from the peer certificate.
+ *
+ * @param filter_envoy_ptr is the pointer to the DynamicModuleNetworkFilter object.
+ * @param result_out is the output pointer to the subject string.
+ * @return the length of the subject string, or 0 if SSL is not available.
+ */
+size_t envoy_dynamic_module_callback_network_filter_get_ssl_subject(
+    envoy_dynamic_module_type_network_filter_envoy_ptr filter_envoy_ptr,
+    envoy_dynamic_module_type_buffer_envoy_ptr* result_out);
+
+// ---------------------------- Filter State Callbacks -------------------------
+
+/**
+ * envoy_dynamic_module_callback_network_set_filter_state_bytes is called by the module to set
+ * filter state with a bytes value. The filter state can be read by other filters in the chain and
+ * can influence routing decisions (e.g., tcp_proxy cluster selection).
+ *
+ * @param filter_envoy_ptr is the pointer to the DynamicModuleNetworkFilter object.
+ * @param key_ptr is the key name for the filter state.
+ * @param key_length is the length of the key.
+ * @param value_ptr is the value to set (bytes).
+ * @param value_length is the length of the value.
+ * @return true if the operation is successful, false otherwise.
+ */
+bool envoy_dynamic_module_callback_network_set_filter_state_bytes(
+    envoy_dynamic_module_type_network_filter_envoy_ptr filter_envoy_ptr,
+    envoy_dynamic_module_type_buffer_module_ptr key_ptr, size_t key_length,
+    envoy_dynamic_module_type_buffer_module_ptr value_ptr, size_t value_length);
```

Comment:
I think envoy_dynamic_module_type_module_buffer should be used per our new style